bots/reads.ts

```diff
@@ -272,30 +293,36 @@ const handleUpdateUrl = async (url: string, ctx: ReadsContext, config: ReadsConf
  *
  */
 export const readsBot = (config: ReadsConfig) => {
-  const { botToken, secretToken: _, webhookUrl: _webhookUrlStr } = config;
+  const { botToken, sectors, secretToken: _, webhookUrl: _webhookUrlStr } = config;
   // const webhookUrl = new URL(webhookUrlStr);
 
   const bot = new Telegraf<ReadsContext>(botToken);
 
   // setup session
-  bot.use(session({ defaultSession: createDefaultSession }))
+  bot.use(session({ defaultSession: createDefaultSession }));
 
   // commands
   bot.command('new', handleNew);
   bot.command('post', handlePost);
   bot.command('preview', replyWithPreview);
   bot.command('setdescription', handleSetDescription);
-  bot.command('setsector', handleSetTaxonomy);
   bot.command('settitle', handleSetTitle);
   bot.command('settype', handleSetTag);
+  bot.command('setsector', async (ctx) => { await handleSetTaxonomy(ctx, config.sectors)});
 
   // actions
   bot.action('new', handleNew);
   bot.action('post', handlePost);
   bot.action('setdescription', handleSetDescription);
-  bot.action('setsector', handleSetTaxonomy);
   bot.action('settitle', handleSetTitle);
   bot.action('settype', handleSetTag);
+  bot.action('setsector', async (ctx) => { await handleSetTaxonomy(ctx, config.sectors)});
+
+  // dynamic actions
+  bot.action(/setsector_(.+)/, async (ctx) => {
+    ctx.session.item.taxonomy = [getSectorLabel(config.sectors, ctx.match[1])];
```

Comment:
need to translate `all` to all the real sectors

Reply:
Ended up just naming it `general` and will transpose on the actual API call